### PR TITLE
[doc] Improve `mps` package description

### LINKED ISSUE #104183: Improve top level package description for MPS package
### 📚 The doc issue

I am new to PyTorch and found page [https://pytorch.org/docs/stable/mps.html](https://pytorch.org/docs/stable/mps.html) where it said "This package enables an interface for accessing MPS backend in python". I did not know what MPS stood for. I think the top level package for an area should expand the acronym and explain something about it.

### Suggest a potential alternative/fix

I suggest changing this section in https://github.com/pytorch/pytorch/blob/main/torch/mps/__init__.py to

`This package enables an interface for accessing MPS (Metal Performance Shaders) backend in Python. MPS is an acceleration backend for MacOS that utilizes GPU for computations.`

I am happy to make this change, if it is agreed upon.

cc @svekars @carljparker @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

### PR DESCRIPTION
**Pull request by davidradl**
Fixes #104183

### CODE REVIEW

**Comment by davidradl** [comment]
I assume this minimal change will not be in the release notes

**Review by malfet — APPROVED** [review]

File: torch/mps/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 r"""
-This package enables an interface for accessing MPS backend in python
+This package enables an interface for accessing MPS (Metal Performance Shaders) backend in Python. MPS is an acceleration backend for MacOS that utilizes GPU for computations.
```
Comment (malfet) [inline torch/mps/__init__.py:2]: Please split the description into several lines and add link to https://developer.apple.com/documentation/metalperformanceshaders

Also, it's not limited to MacOS, i.e. one should be able to use the same on iOS as well.

Comment (davidradl) [reply torch/mps/__init__.py]: @malfet good point about the operating system. I notice the link you provided has a lit of operating systems. I decided to not list the operating systems, as they could change, but instead refer to metal GPUs; this seems more future proof . Let me know if you would prefer a different approach.

**Comment by davidradl** [comment]
Merging after approval

**Pull request closed without merging**